### PR TITLE
Add Gender Class

### LINKED ISSUE #95: Add Gender Class

### PR DESCRIPTION
**Pull request by deeyonn**
Would like to merge this branch containing the added parameter "Gender" to the master branch.

Resolves #95 

Currently failing 3 tests:
1. parse_invalidValue_failure
2. parseCommand_edit
3. parse_multipleRepeatedFields_acceptsLast

Would appreciate any feedback regarding these tests.

### CODE REVIEW

**Review by cliftonfelix — CHANGES_REQUESTED** [review]
AddComandParserTest.java line 150 -> Add GENDER_DESC_BOB
EditCommandParserTest.java 
- line 185 -> remove whole line
- line 187 -> change to BOB

PersonUtil.java line 54 -> change to PREFIX_GENDER

**Review by maryjess — CHANGES_REQUESTED** [review]
Overall, looking good!
Just to make sure no files are left behind, here are the files that I changed, but I notice didn't change here:
- `test/java/seedu/address/model/person/PersonTest.java`
- `test/java/seedu/address/model/person/UniquePersonListTest.java`
- `test/java/seedu/address/model/person/AddressBookTest.java`

**Review by cliftonfelix — CHANGES_REQUESTED** [review]
- Add tests in `test/java/seedu/address/logic/parser/ParserUtilTest.java`
- `PersonTest.java` line 37 to add gender

**Review by cliftonfelix — APPROVED** [review]

**Review by cliftonfelix — APPROVED** [review]
LGTM!

File: src/main/java/seedu/address/model/person/Gender.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents Person's gender in the address book.
+ * Guarantees: immutable; is always valid
+ */
+public class Gender {
+
+    public static final String MESSAGE_CONSTRAINTS =
+        "Gender should only be one word containing alphabetical characters and it should not be blank";
```
Comment (cliftonfelix) [inline src/main/java/seedu/address/model/person/Gender.java:13]: Please change the message constraints according to REGEX validation

**Review by cliftonfelix — CHANGES_REQUESTED** [review]

File: src/test/java/seedu/address/model/person/GenderTest.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        String invalidGenderType = "";
+        assertThrows(IllegalArgumentException.class, () -> new Gender(invalidGenderType));
+    }
+
+    @Test
+    public void isValidGender() {
+        // null gender type
+        assertThrows(NullPointerException.class, () -> Gender.isValidGender(null));
+
+        // invalid university name
```
Comment (maryjess) [inline src/test/java/seedu/address/model/person/GenderTest.java:27]: Minor: Shouldn't the comment here be "invalid gender name"?

File: src/test/java/seedu/address/model/person/GenderTest.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        assertFalse(Gender.isValidGender("")); // empty string
+        assertFalse(Gender.isValidGender(" ")); // spaces only
+        assertFalse(Gender.isValidGender("^")); // only non-alphanumeric characters
+        assertFalse(Gender.isValidGender("male*")); // contains non-alphanumeric characters
+        assertFalse(Gender.isValidGender("12345")); // numbers only
+        assertFalse(Gender.isValidGender("female25")); // alphanumeric characters
+        assertFalse(Gender.isValidGender("alpha male")); // long names
+
+        // valid university name
```
Comment (maryjess) [inline src/test/java/seedu/address/model/person/GenderTest.java:36]: Same here!

**Review by maryjess — APPROVED** [review]
LGTM, would be good if you can apply the changes to the comments even though it is just minor :)

**Review by cliftonfelix — APPROVED** [review]

**Pull request merged**